frame/staking-rewards/src/tests.rs

```diff
@@ -605,11 +708,64 @@ mod claim {
 			configure_default_pica();
 			let stake = 1_000_000_000_000;
 			assert_ok!(<Tokens as Mutate<AccountId>>::mint_into(PICA, &ALICE, stake));
-			let instance_id = <StakingRewards as Staking>::stake(&PICA, &ALICE, stake, WEEK, false)
-				.expect("impossible; qed;");
+			let instance_id =
+				<StakingRewards as Staking>::stake(&PICA, &ALICE, stake, Some(WEEK), false)
+					.expect("impossible; qed;");
 			process_block(REWARD_EPOCH_DURATION_BLOCK);
-			assert_ok!(StakingRewards::claim(Origin::signed(TREASURY), instance_id, TREASURY,));
-			assert_ok!(StakingRewards::claim(Origin::signed(ALICE), instance_id, ALICE,));
+			assert_ok!(StakingRewards::claim(Origin::signed(TREASURY), instance_id, TREASURY));
+			assert_ok!(StakingRewards::claim(Origin::signed(ALICE), instance_id, ALICE));
+		});
+	}
+}
+
+mod usecase {
+	use super::*;
+
+	#[test]
+	fn stake_transfer_reward_claim_unstake() {
+		new_test_ext().execute_with(|| {
+			configure_default_pica();
+			let stake_asset = PICA;
+			let stake = 1_000_000_000_000_u128;
+			let reward_asset = LTC;
+			let reward = 0xC0DEBABE_u128 * 1_000_000_000_000_u128;
+			assert_ok!(<Tokens as Mutate<AccountId>>::mint_into(stake_asset, &ALICE, stake));
+			assert_ok!(<Tokens as Mutate<AccountId>>::mint_into(reward_asset, &TREASURY, reward));
+			let instance_id =
+				<StakingRewards as Staking>::stake(&stake_asset, &ALICE, stake, Some(WEEK), false)
```

Comment:
I see that the tests are missing a case for having multiple fNFTs positions staked earning rewards. This test might be a good place to add it?

Reply:
Another good test to have is two fNFTs having the same reward asset being rewarded but having different staked assets.